rotkehlchen/rotkehlchen.py

```diff
@@ -555,23 +556,82 @@ def get_history_query_status(self) -> Dict[str, str]:
 
     def process_history(
             self,
+            report_id: int,
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> Tuple[Dict[str, Any], str]:
-        (
-            error_or_empty,
-            history,
-            loan_history,
-            asset_movements,
-            eth_transactions,
-            defi_events,
-            ledger_actions,
-        ) = self.events_historian.get_history(
-            start_ts=start_ts,
-            end_ts=end_ts,
-            has_premium=self.premium is not None,
-        )
+        def process_cache(cache_data: List[Dict[str, Any]]) -> Union[None, HistoryResult]:
+            if cache_data:
+                actions = [y for x in cache_data for y in x.values()]
+
+                return (
+                    error_or_empty,
+                    list(filter(lambda x: type(x) in ['Trade',
+                                                      'MarginPosition',
+                                                      'AMMTrade'], actions)),
+                    list(filter(lambda x: type(x) in ['Loan'], actions)),
+                    list(filter(lambda x: type(x) in ['AssetMovement'], actions)),
+                    list(filter(lambda x: type(x) in ['EthereumTransaction'],
+                                actions)),
+                    list(filter(lambda x: type(x) in ['DefiEvent'], actions)),
+                    list(filter(lambda x: type(x) in ['LedgerAction'], actions)))
+            return '', [], [], [], [], [], []
+
+        if report_id:
+            (
+                data,
+                query_start_ts,
+                query_end_ts) = self.data.cache.get_or_query_report_events(
+                report_id=report_id,
+                start_ts=start_ts,
+                end_ts=end_ts)
+            (
+                error_or_empty,
+                history,
+                loan_history,
+                asset_movements,
+                eth_transactions,
+                defi_events,
+                ledger_actions,
+            ) = process_cache(data)
+            if query_start_ts and query_end_ts:
```

Comment:
Let me see If I understand correctly what you meant to do here.

What you wanted to do is:

1. Get all cached data for the given time range.
2. Get the range to query if the cached range does not cover it. This here is tricky, which is why you tried with `query_start_ts` and `query_end_ts` BUT as per the comments I left in `get_or_query_report_events` this won't work.
3. Query external sources to create the data for the missing time ranges

Do you understand from my comments before why this won't work as you have it?